ironic/drivers/modules/drac/management.py

```diff
@@ -308,7 +331,143 @@ class DracRedfishManagement(redfish_management.RedfishManagement):
+                job_id = job_id.strip("'")
+                job_url = '%s/%s' % (jobs_url, job_id)
+                delete_response = requests.delete(
+                    job_url, headers=_HEADERS, verify=False,
+                    auth=(self.redfish_username, self.redfish_password))
+                LOG.info(
+                    'Job queue for iDRAC %s successfully cleared' %
+                    (self.redfish_ip))
+        except drac_exceptions.BaseClientException as exc:
+            LOG.error('DRAC driver failed to clear the job queue for node '
+                      '%(node_uuid)s. Reason: %(error)s.',
+                      {'node_uuid': task.node.uuid, 'error': exc})
+            raise exception.DracOperationError(error=exc)
+
+    @METRICS.timer('DracRedfishManagement.reset_idrac')
+    @base.clean_step(priority=0)
+    def reset_idrac(self, task):
+        """Reset the iDRAC.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        self.get_redfish_info(task)
+        reset_idrac_url = 'https://%s%s' % (self.redfish_ip, _RESET_URL)
+        payload = {"ResetType": "GracefulRestart"}
+
+        try:
+            reset_response = requests.post(
+                reset_idrac_url,
+                data=json.dumps(payload), headers=_HEADERS, verify=False,
+                auth=(self.redfish_username, self.redfish_password))
+
+        except drac_exceptions.BaseClientException as exc:
+            LOG.error('DRAC driver failed to reset idrac for node '
+                      '%(node_uuid)s. Reason: %(error)s.',
+                      {'node_uuid': task.node.uuid, 'error': exc})
+            raise exception.DracOperationError(error=exc)
+
+        LOG.info("Waiting for the iDRAC to become ready")
+        if(self.wait_for_idrac_ready() != True):
+            LOG.error('Timeout reached to become iDRAC ready')
+
+    @METRICS.timer('DracRedfishManagement.known_good_state')
+    @base.clean_step(priority=0)
+    def known_good_state(self, task):
+        """Reset the iDRAC, Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+
+        self.clear_job_queue(task)
+        self.reset_idrac(task)
+
+    @task_manager.require_exclusive_lock
+    def set_boot_device(self, task, device, persistent=False):
+        super(
+            DracRedfishManagement,
+            self).set_boot_device(
+            task,
+            device,
+            persistent=False)
+        if task.node.driver_internal_info.get("clean_steps"):
+            if task.node.driver_internal_info.get("clean_steps")[0].get(
+                    'step') in _CLEAR_JOBS_CLEAN_STEPS:
+                self.clear_job_queue(task)
```

Comment:
We really shouldn't clear the entire job queue in this situation.  The user could be trying to run the reset iDRAC cleaning step, and they wouldn't expect the entire job queue to be cleared as a result of that.  We should only delete the pendng jobs, like we do here:
https://github.com/openstack/ironic/blob/master/ironic/drivers/modules/drac/management.py#L207

Reply:
Addressed, This will execute to delete only unfinished jobs in case of "clear_job_queue" and "known_good_state" clean steps.
If the clean step is only "reset_idrac" then this will not execute, as user wants to reset idrac only.